core/src/main/scala/pureconfig/DurationConvert.scala

```diff
@@ -42,9 +43,11 @@ private[pureconfig] object DurationConvert {
    */
   def fromDuration(d: Duration): String = {
     d match {
+      case f: FiniteDuration => fromFiniteDuration(f)
+      // We must do an `eq` instead of `==` comparison because Undefined is intentionally != itself.
+      case i: Duration.Infinite if i eq Duration.Undefined => Duration.Undefined.toString
       case i: Duration.Infinite if i == Duration.MinusInf => "MinusInf"
       case i: Duration.Infinite => "Inf"
```

Comment:
I also believe that we should encode `Duration.MinusInf`, `Duration.Inf` and `Duration.Undefined` in configs using the same convention. I have a preference for `MinusInf`, `Inf` and `Undefined`, but I'd be ok with encoding them with the leading `Duration.`.